index.html

```diff
@@ -86,50 +109,226 @@
-      options.yearRange = d3.event.target.value;
-      update();
-    });
+    chart.svg.append("g")
+      .append("text")
+        .attr("x", width/2)
+        .attr("y", height+margin.top)
+        .attr("dy", ".71em")
+        .style("text-anchor", "middle")
+        .text("Percent of Workers that are Women");
+
+      var gx = chart.svg.append("g")
+          .attr("class", "x axis")
+          .call(xAxis);
+
+      gx.selectAll("g").filter(function(d) { return d; })
+          .classed("minor", true);
+
+
+    var formatAsDollars = d3.format("$1,000")
+
+    var yAxis = d3.svg.axis()
+                      .scale(chart.y)
+                      .tickFormat(formatAsDollars)
+                      .tickSize(width)
+                      .orient("right")
+                      .ticks(4);
+
+    chart.svg.append("g")
+      .append("text")
+        .attr("transform", "rotate(-90)")
+        .attr("x", -1*(height/2))
+        .attr("y", -1*(width/11))
+        .attr("dy", ".71em")
+        .style("text-anchor", "middle")
+        .text("Median Weekly Earnings");
+
+      var gy = chart.svg.append("g")
+          .attr("class", "y axis")
+          .call(yAxis);
+
+      gy.selectAll("g").filter(function(d) { return d; })
+          .classed("minor", true);
+
+      gy.selectAll("text")
+          .attr("x", -35);
+
+      chart.tooltip2 = d3.tip()
+        .attr('class', 'tooltip')
+        .html(function (d) {return d.SOCName; });
+//        .html(function (d) {return d.SOCName + "<br> Earn " + d.MedEarn; });
+
+      chart.svg.call(chart.tooltip2);
+    
+      chart.tooltip = d3.tip()
+        .attr('class', 'tooltip')
+        .offset([-6,0])
+        .html(function (d) {return d.length + ' Occupations' ; });
+
+      chart.svg.call(chart.tooltip);
+
+     chart.svg.append("g")
+        .append("text")
+        .attr("x", width+5)
+        .attr("y", chart.y(870))
+        .attr("dy", ".71em")
+        .style("text-anchor", "start")
+        .text("Total Median");
+
+     chart.svg.append("g")
+        .append("text")
+        .attr("x", width+5)
+        .attr("y", chart.y(770))
+        .attr("dy", ".71em")
+        .style("text-anchor", "start")
+        .text("Earnings"); 
+
+      chart.update()
   }
 
-  function update() {
+  Chart.prototype.update = function () {
+    var chart = this;
+
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.SOCGroup !== 'all') {
+      var SOCGroup=options.SOCGroup;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.SuperGroup === SOCGroup;
       });
     }
 
+
     // SCATTERPLOT
 
+    var points=chart.svg.selectAll('.point')
+      .data(txData)
+
+      points.enter().append('circle')
+      .attr('class','point')
+//      .attr('class', function(d) { return d.SuperGroup;})
```

Comment:
I'd like to assign each point to a class based on the filter (with different fills for each class) and this method assigns the class, but the the points no longer exit with the filter. I also tried 'point ' + function(d) { return d.SuperGroup;}) but I got the text literal, not the concatenation of point and SuperGroup that I was expecting.